src/parse_ncbi_xml.py

```diff
@@ -1,54 +1,93 @@
-from .util import *
-from Bio.Blast import NCBIXML
 import re
 import sys
-    
+
+from Bio.Blast import NCBIXML
+
+from .util import *
+
+HEADER = ("id,start,end,alignment_id,enzyme_id,enzyme_code,product,"
+"organism,existence,gene,score,evalue,identity_percentage,query_coverage\n")
+
+# csv column title
+HEADER_ELEMENT = ["id", "start", "end", "alignment_id", "enzyme_id",
+    "enzyme_code",  "product", "organism", "existence", "gene", "score",
+    "evalue", "identity_percentage", "query_coverage"]
+HEADER = ",".join(HEADER_ELEMENT)
+
+
+REGEX_ELEMENTS = re.compile("(.*) OS=(.*) OX=.* PE=(\\d)")
+# OS: organism
+# PE: protein existence
+
+REGEX_PRODUCT = re.compile("^(.+)~{3}(.+)~{3}(.+)$")
+REGEX_GENE = re.compile("GN=(.*) PE=")
+
+
+def parse_product_split(labels):
+    # if labeled: {enzyme_id}~~~{enzyme_code}~~~{product}
+    # if not    : {product}
+    labels_split = labels.replace(",", "").split("~~~")
+    try:
+        enzyme_id, enzyme_code, product_name = labels_split[0:4]
```

Comment:
labels_split[0:3]?
Though I've checked that labels_split[0:4] won't raise IndexError.

Reply:
slicing won't cause errors.
raise is caused by assigning results into 3 variables,

Reply:
Now I get it. If it's [0:3], then the potential error (split into more than 3 variables) cannot be detected and caught.
I'll change it back to [0:4].